lsp/src/utils/http/request.ts

```diff
@@ -0,0 +1,32 @@
+
+import { Headers, XHROptions, XHRResponse, xhr } from 'request-light'
+
+
+
+// Abstract away interfaces of existing implementations
+export type HttpRequestOptions = Pick<XHROptions, 'headers'>
+export type HttpRequestHeaders = Headers
+export type HttpResponse = XHRResponse
+
+export interface HttpRequesterI {
```

Comment:
convention is `HttpRequester` for interface and `DefaultHttpRequester` for the concrete implementation